### ISSUE TITLE
CIRI2 :  CIRI_AS_v1.2.pl : Please make sure the SAM is the same one used for circRNA detection by CIRI without modification!

### ISSUE THREAD

**Issue by Thananonsr** [Kevinzjy/CIRIquant-cookbook#35]
Hi everyone
Currently, I am analyzing circRNAs from an RNA-Seq dataset using CIRI2 tools, following the steps outlined in the CIRI-cookbook. Here's how I have proceeded:

1. Checked the FastQC of the raw data files.
2. Trimmed adapters using Trimmomatic and then checked the FastQC.
3. Aligned reads using BWA-MEM to obtain a .sam file.
4. After obtaining the trimmed_sample.sam file, I ran CIRI using the following command:
`perl CIRI2.pl -I trimmed_sample.sam -O sample/trimmed_sample.ciri -F bwa.index_hg38.fa/hg38.fa -A knownGene.gtf`
5. After getting output.ciri, I ran the following command:
`perl CIRI_AS_v1.2.pl -S trimmed_sample.sam -C sample/trimmed_sample.ciri -F bwa.index_hg38.fa/hg38.fa -A knownGene.gtf -O sample/output -D yes`

However, I encountered an issue when I ran the last command. It failed with the error message:
**"Please make sure the SAM is the same one used for circRNA detection by CIRI without modification!"**

My question is:
1. I want to understand what the error **SAM is the same one used for circRNA detection by CIRI without modification!** means and how it occurred, as I haven't made any changes to the .sam file? I've tried to find a solution but haven't been able to fix it, so I would like to ask for your help.

Thank you

**Comment by Kevinzjy** [comment]
Hi @Thananonsr , please make sure you are running CIRI2 & CIRI-AS with paired, untrimmed reads.

**Comment by Thananonsr** [comment]
> Hi @Thananonsr , please make sure you are running CIRI2 & CIRI-AS with paired, untrimmed reads.

Of course, I have been running CIRI2 & CIRI-AS with paired-end trimmed reads.

**Comment by Kevinzjy** [comment]
Yes, that's the problem. CIRI-AS cannot handle reads with different lengths after trimming, so could you try to use untrimmed reads to instread?

**Comment by Thananonsr** [comment]
> Yes, that's the problem. CIRI-AS cannot handle reads with different lengths after trimming, so could you try to use untrimmed reads to instread?

You means use the raw data as a input?

**Comment by Kevinzjy** [comment]
> You means use the raw data as a input?

Yes, only raw data is supported.

**Comment by Thananonsr** [comment]
> > You means use the raw data as a input?
> 
> 
> 
> Yes, only raw data is supported.
> 
> according to other article that I already reviewed also shows and recommended to trimmed reads before run CIRI2. 

That’s why I wonder how low-quality data or adapters in raw data might affect the accuracy and lead to poor results when identifying circRNAs.

**Comment by Thananonsr** [comment]
> > You means use the raw data as a input?
> 
> 
> 
> Yes, only raw data is supported.
> 
> 

According to other articles I have reviewed, it is also recommended to trim the reads before running CIRI2. 

That’s why I wonder how low-quality data or adapters in raw data might affect the accuracy and lead to poor results when identifying circRNAs.

**Comment by Thananonsr** [comment]
Thank you, sir. 
But now I have been running the same step in another dataset (e.g. GSE147009 *RNA-seq data) with raw data. However, the raw data from this database has already had the adapters removed from all samples (24 samples). When I ran the following command:
perl CIRI_AS_v1.2.pl \
-S /home/storeone/sirinar/thananon/SRR11310298/SRR11310298.sam \
-C /home/storeone/sirinar/thananon/SRR11310298/SRR11310298.ciri \
-F bwa.index_hg38.fa/hg38.fa \
-A gencode.v46.annotation.gtf \
-O /home/storeone/sirinar/thananon/SRR11310298 \
-D yes
I reencountered the same issue:
**'Please make sure the SAM is the same one used for circRNA detection by CIRI without modification!'**

**Q: Do you have any suggestions for this issue? I need help since I can't find this database with untrimmed adapters.**

**Comment by Thananonsr** [comment]
And here are all the steps in my analysis:

1. Raw data (RNA-seq)
2. FastQC
3. Index human reference genome
4. Align reads of the sample with BWA MEM -T 19
5. Run perl CIRI2.pl to detect circRNAs
6. Run perl CIRI_AS
7. Running the CIRI-full pipeline
8. Running CIRI-vis
9. circRNA quantification (CIRIquant)

I followed the above steps for circRNA analysis per sample, and the resulting data is in GTF format.

--> I’m wondering if it's necessary to run all the steps from 5-9. I’ve already obtained circRNA data from step 5, but I’m unable to run step 6.
**Q: 
**1. Can I take the circRNA data from step 5 and skip directly to step 9? Or would this negatively affect the circRNA data output?****

**Comment by Kevinzjy** [comment]
> Thank you, sir. But now I have been running the same step in another dataset (e.g. GSE147009 *RNA-seq data) with raw data. However, the raw data from this database has already had the adapters removed from all samples (24 samples). When I ran the following command: perl CIRI_AS_v1.2.pl -S /home/storeone/sirinar/thananon/SRR11310298/SRR11310298.sam -C /home/storeone/sirinar/thananon/SRR11310298/SRR11310298.ciri -F bwa.index_hg38.fa/hg38.fa -A gencode.v46.annotation.gtf -O /home/storeone/sirinar/thananon/SRR11310298 -D yes I reencountered the same issue: **'Please make sure the SAM is the same one used for circRNA detection by CIRI without modification!'**
> 
> **Q: Do you have any suggestions for this issue? I need help since I can't find this database with untrimmed adapters.**

Hi @Thananonsr, you can trim the reads to a uniform length (e.g. 75/100bp) to run CIRI-full.
